crates/quantum_info/src/clifford.rs

```diff
@@ -0,0 +1,247 @@
+    }
+
+    /// Modifies the tableau in-place by appending SX-gate
+    pub fn append_sx(&mut self, qubit: usize) {
+        let (mut x, z, mut p) = self.tableau.multi_slice_mut((
+            s![.., qubit],
+            s![.., self.num_qubits + qubit],
+            s![.., 2 * self.num_qubits],
+        ));
+
+        azip!((p in &mut p, &x in &x, &z in &z)  *p ^= !x & z);
+        azip!((&z in &z, x in &mut x) *x ^= z);
+    }
+
+    /// Modifies the tableau in-place by appending SXDG-gate
+    pub fn append_sxdg(&mut self, qubit: usize) {
+        let (mut x, z, mut p) = self.tableau.multi_slice_mut((
+            s![.., qubit],
+            s![.., self.num_qubits + qubit],
+            s![.., 2 * self.num_qubits],
+        ));
+
+        azip!((p in &mut p, &x in &x, &z in &z)  *p ^= x & z);
+        azip!((&z in &z, x in &mut x) *x ^= z);
+    }
+
+    /// Modifies the tableau in-place by appending H-gate
+    pub fn append_h(&mut self, qubit: usize) {
+        let (mut x, mut z, mut p) = self.tableau.multi_slice_mut((
+            s![.., qubit],
+            s![.., self.num_qubits + qubit],
+            s![.., 2 * self.num_qubits],
+        ));
+
+        azip!((p in &mut p, &x in &x, &z in &z)  *p ^= x & z);
+        azip!((x in &mut x, z in &mut z)  (*x, *z) = (*z, *x));
+    }
+
+    /// Modifies the tableau in-place by appending SWAP-gate
+    pub fn append_swap(&mut self, qubit0: usize, qubit1: usize) {
+        let (mut x0, mut z0, mut x1, mut z1) = self.tableau.multi_slice_mut((
+            s![.., qubit0],
+            s![.., self.num_qubits + qubit0],
+            s![.., qubit1],
+            s![.., self.num_qubits + qubit1],
+        ));
+        azip!((x0 in &mut x0, x1 in &mut x1)  (*x0, *x1) = (*x1, *x0));
+        azip!((z0 in &mut z0, z1 in &mut z1)  (*z0, *z1) = (*z1, *z0));
+    }
+
+    /// Modifies the tableau in-place by appending CX-gate
+    pub fn append_cx(&mut self, qubit0: usize, qubit1: usize) {
+        let (x0, mut z0, mut x1, z1, mut p) = self.tableau.multi_slice_mut((
+            s![.., qubit0],
+            s![.., self.num_qubits + qubit0],
+            s![.., qubit1],
+            s![.., self.num_qubits + qubit1],
+            s![.., 2 * self.num_qubits],
+        ));
+        azip!((p in &mut p, &x0 in &x0, &z0 in &z0, &x1 in &x1, &z1 in &z1) *p ^= (x1 ^ z0 ^ true) & z1 & x0);
+        azip!((x1 in &mut x1, &x0 in &x0) *x1 ^= x0);
+        azip!((z0 in &mut z0, &z1 in &z1) *z0 ^= z1);
+    }
+
+    /// Modifies the tableau in-place by appending W-gate.
+    /// This is equivalent to an Sdg gate followed by an H gate.
+    pub fn append_v(&mut self, qubit: usize) {
+        let (mut x, mut z) = self
+            .tableau
+            .multi_slice_mut((s![.., qubit], s![.., self.num_qubits + qubit]));
+
+        azip!((x in &mut x, z in &mut z) (*x, *z) = (*x ^ *z, *x));
+    }
+
+    /// Modifies the tableau in-place by appending CZ-gate
+    pub fn append_cz(&mut self, i: usize, j: usize) {
```

Comment:
I think it's easy to write it explicitly, see: https://github.com/Qiskit/qiskit/blob/eb8f791649719867bac19d3371a7dcb8c404fcc4/qiskit/quantum_info/operators/symplectic/clifford_circuits.py#L422

Reply:
Thanks. I have updated this and some other gates based on the python code. There are still a few gates (namely CY, ECR, iSWAP, DCX) where we don't have an explicit tableau manipulation (neither in the python or the new rust code), but we can leave this as a follow-up.